perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -44,13 +47,54 @@ enum Flags
     Flag_count
 };
 
-static void SetFlags(u16 Result, bool* FlagArray)
-{
-    FlagArray[Flag_SF] = Result & 0x8000;
-    FlagArray[Flag_ZF] = (Result == 0);
-}
+static void SetFlags(const instruction& Instruction, const u16 LeftOperandValue, 
+    const u16 RightOperandValue, const u16 Result, bool* FlagArray)
+{   
+    u16 HighOrderBit = (Instruction.Operands[1].Register.Count == 1) ? 0x80 : 0x8000;
+    bool Parity = true;
+       
+    switch (Instruction.Op)
+    {
+        case Op_add:
+        {
+            FlagArray[Flag_CF] = ((LeftOperandValue & HighOrderBit) || (RightOperandValue & HighOrderBit))
+                && !(Result & HighOrderBit);
+            FlagArray[Flag_AF] = ((LeftOperandValue & 0x8) || (RightOperandValue & 0x8))
+                && !(Result & 0x8);
+            FlagArray[Flag_OF] = (!((LeftOperandValue ^ RightOperandValue) & HighOrderBit))
+                && ((Result ^ LeftOperandValue) & HighOrderBit);
+            FlagArray[Flag_SF] = Result & HighOrderBit;
+            FlagArray[Flag_ZF] = (Result == 0);
+            for (u16 i = 0; i < 8; i++)
+            {
+                if (Result & (1 << i))
+                {
+                    Parity = !Parity;
+                }
+            }
+            FlagArray[Flag_PF] = Parity;
+        } break;
+        case Op_sub: [[fallthrough]];
```

Comment:
This case can be removed, it does nothing here. It will go into default whether it's here or not.

Reply:
Left it because there's an actual default now
3dd47bb7df7996fefafadf6528ffbbd2c60da158